1-js/06-advanced-functions/05-global-object/article.md

```diff
@@ -1,155 +1,87 @@
 
-1. Using global variables is generally discouraged. There should be as few global variables as possible, but if we need to make something globally visible, we may want to put it into `window` (or `global` in Node.js).
+## Usando para polyfills
 
-    Here we put the information about the current user into a global object, to be accessible from all other scripts:
+Usamos o objeto global para testar o suporte aos recursos modernos da linguagem.
 
-    ```js run
-    // explicitly assign it to `window`
-    window.currentUser = {
-      name: "John",
-      age: 30
-    };
+Por exemplo, testar se o objeto interno `Promise` existe (ele não existe em navegadores antigos):
+```js run
+if (!window.Promise) {
+  alert("Seu navegador é muito antigo!");
+}
+```
 
-    // then, elsewhere, in another script
-    alert(window.currentUser.name); // John
-    ```
+Se não houver (vamos dizer que estamos em um navegador antigo), podemos criar "polyfills": adicionar funções que não são suportadas pelo ambiente, mas existem no padrão moderno.
 
-2. We can test the global object for support of modern language features.
+```js run
+if (!window.Promise) {
+  window.Promise = ... // implementação customizada do recurso moderno da linguagem
+}
+```
 
-    For instance, test if a build-in `Promise` object exists (it doesn't in really old browsers):
-    ```js run
-    if (!window.Promise) {
-      alert("Your browser is really old!");
-    }
-    ```
+## Resumo
 
-3. We can create "polyfills": add functions that are not supported by the environment (say, an old browser), but exist in the modern standard.
+- O objeto global contém variáveis que devem estar disponíveis em qualquer lugar.
 
-    ```js run
-    if (!window.Promise) {
-      window.Promise = ... // custom implementation of the modern language feature
-    }
-    ```
+    Isso inclui objetos internos Javascript, como `Array` e valores específicos do ambiente, como `window.innerHeight` -- a altura da janela no navegador.
+- O objeto global tem o nome universal `globalThis`.
 
-...And of course, if we're in a browser, using `window` to access browser window features (not as a global object) is completely fine.
+    ...Porém é mais frequentemente referido pelos seu nomes específicos de ambientes "old-school", como `window` (navegador) e `global` (Node.js). Como `globalThis` é uma proposta recente, não é suportado pelo não Chromium Edge (mas pode ser usado com um polyfill).
+- Devemos salvar valores no objeto global apenas se eles forem realmente globais em nosso projeto. E manter sua quantidade no mínimo.
+- No navegador, ao menos que estejamos usando [modules](info:modules), funções e variáveis globais declaradas com `var` se tornar uma propriedade do objeto global.
+- Para tornar nosso código à prova de mudanças no futuro e mais fácil de entender, devemos acessar as propriedades diretamente do objeto global, como `window.x`.
```

Comment:
I think it would be better to rephrase it: "acessar as propriedades do objeto global diretamente, como...". The way it is, using "diretamente" after "propriedades" kind of changes the meaning of it. What do you think?